src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -752,13 +753,13 @@ private final static boolean isVCFStream(final InputStream stream, final String
         return new LazyGenotypesContext.LazyData(genotypes, header.getSampleNamesInOrder(), header.getSampleNameToOffset());
     }
 
-    private final String[] INT_DECODE_ARRAY = new String[10000];
+    private final Vector<String> INT_DECODE_VECTOR = new Vector<String>(10000);
```

Comment:
I imagine that with potentially millions, it's possible that the allocation could get large, especially if we start with a capacity of 0.  I attempted to make as little additional overhead as possible.

Good point on the ArrayList vs. Vector; most recent update incorporates this change.  Can you tell I come from C++-land :)?